hoomd/mpcd/CollisionMethod.cc

```
@@ -48,15 +48,272 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+                                        access_location::host,
+                                        access_mode::readwrite);
+    ArrayHandle<Scalar4> h_angmom_accum(m_angmom_accum,
+                                        access_location::host,
+                                        access_mode::readwrite);
+    ArrayHandle<unsigned int> h_embed_group(m_embed_group->getIndexArray(),
+                                            access_location::host,
+                                            access_mode::read);
+    ArrayHandle<Scalar4> h_postype(m_pdata->getPositions(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<Scalar4> h_orientation(m_pdata->getOrientationArray(),
+                                       access_location::host,
+                                       access_mode::read);
+    ArrayHandle<Scalar4> h_velocity(m_pdata->getVelocities(),
+                                    access_location::host,
+                                    access_mode::read);
+    ArrayHandle<unsigned int> h_body(m_pdata->getBodies(),
+                                     access_location::host,
+                                     access_mode::read);
+    ArrayHandle<unsigned int> h_rtag(m_pdata->getRTags(), access_location::host, access_mode::read);
+    for (unsigned int idx = 0; idx < num_group; ++idx)
+        {
+        // get the index from the embedded group
+        unsigned int particle_index = h_embed_group.data[idx];
+        // check if particle is in a rigid body
+        unsigned int central_tag = h_body.data[particle_index];
+        if (central_tag >= MIN_FLOPPY)
+            {
+            continue;
+            }
+        // if the central particle is not local, cannot read or write to it.
+        unsigned int central_idx = h_rtag.data[central_tag];
+        if (central_idx == NOT_LOCAL)
+            {
+            continue;
+            }
```

This seems like it shouldn't be here. If the central particle isn't on our rank, it is a fatal error not something we can silently ignore!

---

This check is similar to one in Force Composite, and it has this comment on this check

```
// If the central particle is not local, then we cannot update the position and orientation
// of this particle. Ideally, this would perform an error check. However, that is not
// feasible as ForceComposite does not have knowledge of which ghost particles are within
// the interaction ghost width (and need therefore need to be updated) vs those that are
// communicated to make bodies whole.
```
Is this also true of CollisionMethod, or can we put an error there?

---

I think that comment is talking about something different. I would go ahead and remove this check because we expect the communicator to handle rigid bodies correctly. If the central particle were every not present, there will probably be a segfault.

Alternatively, we could add an `assert(central_idx != NOT_LOCAL);`, which is a debug statement that does not actually run in production code.